shared/src/ledger/eth_bridge/vp/store.rs

```diff
@@ -0,0 +1,92 @@
+//! Functionality for reading from storage in order to validate transactions
+
+use borsh::BorshDeserialize;
+use eyre::{Context, Result};
+
+use crate::ledger::native_vp::Ctx;
+use crate::ledger::storage as ledger_storage;
+use crate::ledger::storage::StorageHasher;
+use crate::types::storage::Key;
+use crate::vm::WasmCacheAccess;
+
+/// Read pre/post storage
+pub(super) trait Reader {
```

Comment:
Since I've added these `read_pre_value` and `read_post_value` to the `Ctx` type in the bridge pool PR, let's just use those instead (just steal them). They return a simpler type (`Option<T>` instead of `Result<Option<T>>`) so it is simpler to handle.

Reply:
We should share code functionality, though I'd rather we make an issue to update the `Ctx` API in `main` and then refactor everywhere at once. We shouldn't silently ignore errors though if they happen (i.e. we should use `Result<Option<T>>` rather than `Option<T>`).

Reply:
These methods are in `eth_bridge_integration` now, so lets use them. But I agree, we shouldn't siliently ignore errors. It should be super fast to refactor the ctx methods to return the `Result<Option<T>>`

Reply:
sure I can refactor there in our codebase quickly, I will make an issue before merging this PR to update the `Ctx` API in `main`. I need to look closer at why it returns `Result<Option<T>>` and not just `Result<T>` as at a glance it looks like these `read_pre`/`read_post` methods error if a key isn't present in storage, I would have thought in that case it would have defaulted to `None` and the errors would be reserved for things like e.g. rocksdb or OS-level errors.